### PR TITLE
[UI] Truncate run id in the run name when comparing multiple runs

### PR DESCRIPTION
**Pull request by harupy**
## What changes are proposed in this pull request?

This PR truncates run id in the run name when comparing multiple runs in order to:
- Prevent the legends from narrowing the plot area
- Make the popover compact

Before
Screen Shot 2020-02-29 at 12 38 01

After
Screen Shot 2020-02-29 at 12 36 51

## How is this patch tested?

Manually tested

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s) does this PR affect?

- [ ] UI
- [ ] CLI
- [ ] API
- [ ] REST-API
- [ ] Examples
- [ ] Docs
- [ ] Tracking
- [ ] Projects
- [ ] Artifacts
- [ ] Models
- [ ] Scoring
- [ ] Serving
- [ ] R
- [ ] Java
- [ ] Python

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/server/js/src/utils/Utils.js [hunk 1]
```diff
@@ -290,7 +290,7 @@ class Utils {
    * @param runTags Object of tag name to MlflowMessages.RunTag instance
    */
   static getRunDisplayName(runTags, runUuid) {
-    return Utils.getRunName(runTags) || "Run " + runUuid;
+    return Utils.getRunName(runTags) || "Run " + runUuid.slice(0, 6);
```
Comment (mparkhe) [inline mlflow/server/js/src/utils/Utils.js]: Does it make sense to show trailing "..." so this truncation is not confusing.
Thoughts on this: @gioa and @smurching

Comment (gioa) [reply mlflow/server/js/src/utils/Utils.js]: Are the shorten run IDs distinct enough? I am worried they show up the same for different runs. 

Agree with @mparkhe, maybe we should truncate it with text-overflow: ellipsis. I also expect users hover can see the full string, or do a triple-click can copy the whole thing.

Comment (harupy) [reply mlflow/server/js/src/utils/Utils.js]: @mparkhe.
Agree on showing trailing `...`.

@gioa 
> Are the shorten run IDs distinct enough?

I think they are.

> I also expect users hover can see the full string, or do a triple-click can copy the whole thing.

What is the use case of this feature?

Comment (gioa) [reply mlflow/server/js/src/utils/Utils.js]: Doesn't the user need the full run IDs to refer to in the API calls? So they may copy it right there? I might be wrong though. 

Also when I see a trailing ..., I would expect to be able to copy the full string, just a pattern I get used to in other places.

Comment (harupy) [reply mlflow/server/js/src/utils/Utils.js]: How about putting a button on the popover that copies the full run id to the user's clipboard?

Comment (harupy) [reply mlflow/server/js/src/utils/Utils.js]: Example:

copy-button

Comment (gioa) [reply mlflow/server/js/src/utils/Utils.js]: That works too! Might be a little overkill. Could add a run ID in the top info section of a run detail page and support copy there.

Comment (harupy) [reply mlflow/server/js/src/utils/Utils.js]: @Zangr Any thoughts?

**Comment by harupy** [comment]
Added trailing `...`.

Screen Shot 2020-03-03 at 23 22 54

File: mlflow/server/js/src/utils/Utils.js (hunk 1 above)
Comment (harupy) [reply mlflow/server/js/src/utils/Utils.js]: Tried a copy button on the popover.

overkill

Comment (gioa) [reply mlflow/server/js/src/utils/Utils.js]: What do you think? It seems too much going on in this popup. I am okay without it. 

Btw I was talking about adding this in the top info section area. See a mock here:
MLFlow_RunDetail (1)

Comment (harupy) [reply mlflow/server/js/src/utils/Utils.js]: Yes, it's too much and confusing. The mock looks great!

File: mlflow/server/js/src/components/MetricsPlotPanel.js [hunk 2]
```diff
@@ -521,7 +521,10 @@ const mapStateToProps = (state, ownProps) => {
   // Flat array of all metrics, with history and information of the run it belongs to
   // This is used for underlying MetricsPlotView & predicting chartType for MetricsPlotControls
   const metricsWithRunInfoAndHistory = _.flatMap(runUuids, (runUuid) => {
-    const runDisplayName = Utils.getRunDisplayName(getRunTags(runUuid, state), runUuid);
+    const runDisplayName = Utils.getRunDisplayName(
+      getRunTags(runUuid, state),
+      runUuid.slice(0, 6) + '...'
```
Comment (Zangr) [inline mlflow/server/js/src/components/MetricsPlotPanel.js]: What do you think about setting the run name wrapper to `text-overflow: ellipsis`? That way, we won't lose any information in the DOM.
https://developer.mozilla.org/en-US/docs/Web/CSS/text-overflow

Comment (harupy) [reply mlflow/server/js/src/components/MetricsPlotPanel.js]: @Zangr Thanks for the comment! If we use `text-overflow`, do we need to set `max-width`?

**Comment by aarondav** [comment]
I like the change - what's the current state of this PR?

File: mlflow/server/js/src/components/MetricsPlotPanel.js (hunk 2 above)
Comment (harupy) [reply mlflow/server/js/src/components/MetricsPlotPanel.js]: @Zangr It seems this can't be applied to the legends.

**Comment by aarondav** [comment]
@Zangr Mind reviewing based on @harupy's latest update?

File: mlflow/server/js/src/experiment-tracking/components/MetricsPlotPanel.js [hunk 3]
```diff
@@ -523,7 +523,10 @@ const mapStateToProps = (state, ownProps) => {
   // Flat array of all metrics, with history and information of the run it belongs to
   // This is used for underlying MetricsPlotView & predicting chartType for MetricsPlotControls
   const metricsWithRunInfoAndHistory = _.flatMap(runUuids, (runUuid) => {
-    const runDisplayName = Utils.getRunDisplayName(getRunTags(runUuid, state), runUuid);
+    const runDisplayName = Utils.getRunDisplayName(
+      getRunTags(runUuid, state),
+      runUuid.slice(0, 6) + '...'
```
Comment (Zangr) [inline mlflow/server/js/src/experiment-tracking/components/MetricsPlotPanel.js:528]: We were using `Utils.truncateString(runDisplayName, MAX_RUN_NAME_DISPLAY_LENGTH)`, could you please check if you can achieve this by use and decrease MAX_RUN_NAME_DISPLAY_LENGTH?

**Comment by aarondav** [comment]
@harupy Mind addressing Richard's feedback?

**Comment by harupy** [comment]
Decreasing `MAX_RUN_NAME_DISPLAY_LENGTH` seems to work. Thanks @Zangr!

Before (MAX_RUN_NAME_DISPLAY_LENGTH = 36):

Screen Shot 2020-04-26 at 17 07 18

After (MAX_RUN_NAME_DISPLAY_LENGTH = 24):

Screen Shot 2020-04-26 at 17 06 41

**Review by Zangr — APPROVED** [review]
LGTM

**Pull request merged**